### PR TITLE
Configuration type inconsistencies

### PR DESCRIPTION
**Pull request by sdimitris**
Most of configuration types was 'any' types. So in order to avoid that, in each module a Config Type  was made.
Each module passes it's type in the moduleManager.

**What kind of change does this PR introduce?** 

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other (please describe)

### CODE REVIEW

**Review by Michael-Vol — APPROVED** [review]

File: modules/authentication/src/interfaces/EmailConfig.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    },
+    mandrill: {
+      apiKey: string;
+    },
+    sendgrid: {
+      apiKey: string;
+      apiUser: string;
+    },
+  },
+};
```
Comment (kon14) [inline modules/authentication/src/interfaces/EmailConfig.ts:29]: Add newline at EOF

File: modules/email/src/email-provider/transports/sendgrid/sendgrid.config.ts [hunk 2]
```diff
@@ -1,6 +1,4 @@
 export interface SendGridConfig {
-
     apiKey: string;
-
 }
```
Comment (kon14) [inline modules/email/src/email-provider/transports/sendgrid/sendgrid.config.ts:4]: Add newline at EOF

File: modules/email/src/admin/admin.ts [hunk 3]
```diff
@@ -53,23 +54,23 @@ export class AdminHandlers {
       });
   }
 
-  private getRegisteredRoutes(): any[] {
+  private static getRegisteredRoutes(): any[] {
```
Comment (kon14) [inline modules/email/src/admin/admin.ts:57]: Are we gonna do this across the board though?

**Review by kon14 — CHANGES_REQUESTED** [review]

File: modules/email/src/email-provider/transports/sendgrid/sendgrid.config.ts (hunk 2 above)
Comment (kon14) [reply modules/email/src/email-provider/transports/sendgrid/sendgrid.config.ts:4]: Idk what's up with GitHub's diff checker, ignore this.

**Review by kon14 — APPROVED** [review]

**Pull request merged**